newsroom/reports/views.py

```diff
@@ -1,33 +1,48 @@
+from pydantic import BaseModel
 from io import StringIO
 import csv
 
 from quart_babel import gettext
 
 from superdesk.core import get_current_app, get_app_config
+from superdesk.core.types import Request
+from superdesk.core.web import EndpointGroup
 from superdesk.flask import session, jsonify, render_template, abort
```

Comment:
It would be good to replace flask functionality with our own. For example (for demonstration purposes, not reflecting any code in this file):

Using Flask
```python
from superdesk.flask import session, jsonify, abort

@blueprint.endpoint("/some/endpoint")
async def some_endpoint(request: Request):
    user_type = session.get("user_type")
    if user_type != "administrator":
        abort(403, "Not allowed")
    return jsonify({"data": {"enabled": True}})
```

Using our framework:
```python
from newsroom.auth.utils import get_user_from_request

@blueprint.endpoint("/some/endpoint")
async def some_endpoint(request: Request) -> Response:
    # `None` can be supplied instead of `request`
    # and the system will obtain the current request instance automatically
    user = get_user_from_request(request)
    
    if user.user_type != "administrator":
        return await request.abort(403, "Not allowed")
    return Response({"data": {"enabled": True}})
```